packages/flutter/lib/src/material/text_selection_toolbar.dart

```diff
@@ -513,57 +536,145 @@ class _RenderTextSelectionToolbarItemsLayout extends RenderBox
   // Decide which children will be painted, set their shouldPaint, and set the
   // offset that painted children will be placed at.
   void _placeChildren() {
-    int i = -1;
     Size nextSize = Size.zero;
-    double fitWidth = 0.0;
     final RenderBox navButton = firstChild!;
-    double overflowHeight = overflowOpen && !isAbove ? navButton.size.height : 0.0;
-    visitChildren((RenderObject renderObjectChild) {
-      i++;
-
-      final RenderBox child = renderObjectChild as RenderBox;
-      final ToolbarItemsParentData childParentData = child.parentData! as ToolbarItemsParentData;
+    final bool isRtl = textDirection == TextDirection.rtl;
 
-      // Handle placing the navigation button after iterating all children.
-      if (renderObjectChild == navButton) {
-        return;
+    if (!overflowOpen) {
```

Comment:
Would it be useful to split the layout when overflowOpen true/false into two separate methods in order to reduce some of the nested conditionals in this method? I find it somewhat hard to read (though it is harder to read on GitHub than in the IDE).